Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h

```diff
@@ -101,7 +102,13 @@ DWORD WINAPI MDD_serialPortReceivingThread(LPVOID p_serial) {
                         if (x > serial->bufferSize) {
                             x = (DWORD)serial->bufferSize;
                         }
-                        ret = ReadFile(serial->hComm, serial->receiveBuffer, x, &serial->receivedBytes, &rdSync);
+                        ret = ReadFile(serial->hComm, &serial->receiveBuffer[idx], x, &serial->receivedBytes, &rdSync);											
+                        idx = idx + serial->receivedBytes;
+                        if(idx == serial->bufferSize)
+                        {
+                          memcpy(serial->completeMessage, serial->receiveBuffer, serial->bufferSize);
+                          idx = 0;
```

Comment:
The message comes in fractions and is stored byte per byte in the receiveBuffer. If bufferSzze bytes have been read, the message is complete used for the packager functions to unpack it. The message fractions don't necessarily come in accordance with the datatypes I want to unpack. If MDD_serialPortReadP is called receiveBuffer might be a mixture of the last and the current message. completeMessage always contains an associated message.